src/jaegertracing/TracerFactoryTest.cpp

```diff
@@ -65,6 +67,18 @@ TEST(TracerFactory, testValidConfig)
         "refreshInterval": 60
     }
   })";
+    testutils::EnvVariable::setEnv("JAEGER_SERVICE_NAME", "");
+    TracerFactory tracerFactory;
+    std::string errorMessage;
+    auto tracerMaybe = tracerFactory.MakeTracer(config, errorMessage);
+    ASSERT_EQ(errorMessage, "");
+    ASSERT_TRUE(tracerMaybe);
+}
+
+TEST(TracerFactory, testEnvironmentConfig)
```

Comment:
Could you add a test with both env var `JAEGER_SERVICE_NAME` and yaml config `service_name` set, to show which one takes precedence.

Reply:
@objectiser done